lib/Data/Database.php

```diff
@@ -199,12 +199,31 @@ public function create($pasteid, array $paste)
             $burnafterreading = $paste['adata'][3];
         }
         try {
+            $big_string       = $isVersion1 ? $paste['data'] : Json::encode($paste);
+            $metajson         = Json::encode($meta);
+            if (self::$_type === 'oci') {
+                // It is not possible to execute in the normal way if strlen($big_string) >= 4000
```

Comment:
Why? Can you include any link to a resource? (And only for OCI DBs is not it?)

And if that happens, then what happens? The fallback below is executed I guess? Which also fails…

Also, what does this mean for actually using PrivateBin with OCI? E.g. does that mean you have a limit of what you can store (likely the file size limit can be imposed here then)?

Reply:
Using PrivateBin with database backend will limit the paste size allowed to be stored based on the chosen data type. For all supported database backends, we try to use the largest allowed types, in some of these it is configurable:
- MySQL BLOB limited by [max_allowed_packet, defaults to 64 MiB](https://dev.mysql.com/doc/refman/8.0/en/server-system-variables.html#sysvar_max_allowed_packet)
- Oracle [CLOB 4 MiB](https://docs.oracle.com/database/timesten-18.1/TTPLP/d_lob.htm#TTPLP66609)
- PostgreSQL [TEXT 1 GiB](https://wiki.postgresql.org/wiki/FAQ#What_is_the_maximum_size_for_a_row.2C_a_table.2C_and_a_database.3F)
- SQLite [BLOB 1GB](https://www.sqlite.org/limits.html#max_length)

Reply:
[CLOB is 4 GB](https://docs.oracle.com/cd/B19306_01/server.102/b14237/limits001.htm#i287903), you quoted the wrong thing.